### PR TITLE
Add a proto.to_any method

### PR DESCRIPTION
**Pull request by idiamond-stripe**
This PR uses the code from https://github.com/stripe/skycfg/pull/46 and responds to the review feedback on this PR.

I've removed the gogo code entirely to be consistent with the rest of the provided functions in `proto_api.go`, which don't support gogo protos.

This PR is also quite old so I've also rebased it onto master.

### CODE REVIEW

**Comment by idiamond-stripe** [comment]
r? @jmillikin-stripe 

The build is failing but I'm confused as to why given the dependencies don't appear to have changed.

File: internal/go/skycfg/proto_api.go [hunk 1]
```diff
@@ -25,8 +25,9 @@ import (
 
 	"github.com/golang/protobuf/jsonpb"
 	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
 	"go.starlark.net/starlark"
-	yaml "gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v2"
```
Comment (jmillikin-stripe) [inline internal/go/skycfg/proto_api.go:30]: Can you keep this import alias as-is? It's nice to have explicitly named imports when upstream uses a non-default package name.

File: internal/go/skycfg/proto_api.go [hunk 2]
```diff
@@ -236,6 +238,30 @@ func fnProtoToJson(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple
...
+// Implementation of the `proto.to_any()` built-in function. Returns a
+// skyProtoMessage with an `Any` proto.Message in it.
+func fnProtoToAny(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var msg *skyProtoMessage
+	err := wantSingleProtoMessage(fn.Name(), args, kwargs, &msg)
+	if err != nil {
+		return nil, err
+	}
+
+	// Disambiguate between golang encoded proto messages.
```
Comment (jmillikin-stripe) [inline internal/go/skycfg/proto_api.go:250]: I'm not sure why this branch is needed. Why not `any, err := ptypes.MarshalAny(msg.msg)`? Was there a case you encountered where a protobuf message has no type name?

**Review by jmillikin-stripe — CHANGES_REQUESTED** [review]
Looks generally good -- two minor comments.

**Comment by jmillikin-stripe** [comment]
> The build is failing but I'm confused as to why given the dependencies don't appear to have changed.

It looks like something in `Go: master` is unhappy with the contents of `go.mod`. No idea why, but since it's not affecting any stable release I think it's fine to ignore.

We should probably drop `Go: master` from the CI suite since I don't see an obvious way to make Travis test failures informational.

**Comment by dilyevsky** [comment]
Likely remnant of the pr this is based on - it used this check to
differentiate between gogo and standard registries (which I’m not sure if
they are actually equivalent or not)

File: internal/go/skycfg/proto_api.go (hunk 2 above)
Comment (idiamond-stripe) [reply internal/go/skycfg/proto_api.go:250]: I think this is a holdover/cargo-cult from the first iteration of this change. I'll remove it.

File: internal/go/skycfg/proto_api.go (hunk 1 above)
Comment (idiamond-stripe) [reply internal/go/skycfg/proto_api.go:30]: I think this is a holdover/cargo-cult from the first iteration of this change. Let me change the alias back.

**Comment by idiamond-stripe** [comment]
ptal @jmillikin-stripe

**Pull request merged**